tools/clang/lib/CodeGen/CGDebugInfo.cpp

```diff
@@ -1047,8 +1047,15 @@ bool CGDebugInfo::TryCollectHLSLRecordElements(const RecordType *Ty,
     unsigned VecSize = hlsl::GetHLSLVecSize(QualTy);
     unsigned ElemSizeInBits = CGM.getContext().getTypeSize(ElemQualTy);
     unsigned CurrentAlignedOffset = 0;
+    std::string FieldNameBuf;
     for (unsigned ElemIdx = 0; ElemIdx < VecSize; ++ElemIdx) {
-      StringRef FieldName = StringRef(&"xyzw"[ElemIdx], 1);
+      StringRef FieldName;
+      if (VecSize <= 4) {
```

Comment:
This fixes OOB accesses into the constant string for long vectors. Asan was reporting an error for the tools\clang\test\CodeGenDXIL\hlsl\intrinsics\linalg-builtins.hlsl test:
```
==214895==ERROR: AddressSanitizer: global-buffer-overflow on address 0x7f9d9299a905 at pc 0x7f9d8fd110be bp 0x7ffcc6fe4b40 sp 0x7ffcc6fe4b38
READ of size 1 at 0x7f9d9299a905 thread T0
```

Reply:
Pulled out this fix into #7332 and reverted the change here.
This means this PR will see get asan test failures in the Azure Pipelines until #7332 lands